packages/blade/src/components/Input/TextInput/TextInput.tsx

```diff
@@ -217,6 +222,13 @@ const _TextInput: React.ForwardRefRenderFunction<BladeElementRef, TextInputProps
 ): ReactElement => {
   const textInputRef = useBladeInnerRef(ref);
   const [shouldShowClearButton, setShouldShowClearButton] = useState(false);
+  const isLabelDefined = typeof label === 'string' && label.length > 0;
+  const isAccessibilityLabelDefined =
+    typeof accessibilityLabel === 'string' && accessibilityLabel.length > 0;
+
+  if (!(isLabelDefined || isAccessibilityLabelDefined)) {
+    throw new Error('[Blade TextInput]: Either label or accessibilityLabel is required');
+  }
```

Comment:
Lets reduce runtime checks as much as possible, most of the time we do runtime checks for things which can't be easily done with TypeScript. 

But this label vs accessibilityLabel case can be done via TypeScript, we are doing the exact same thing in BaseButton also with children vs icon. 


```ts

type TextInputCommonProps = Pick<
  BaseInputProps,
  ...
>

/*
  Mandatory accessibilityLabel prop when label is not provided
*/
type TextInputPropsWithLabel = {
  /**
   * Label to be shown for the input field
   */
  label?: undefined;
  /**
   * Accessibility label for the input
   */
  accessibilityLabel: string;
};

/*
  Optional accessibilityLabel prop when label is provided
*/
type TextInputPropsWithA11yLabel = {
  /**
   * Label to be shown for the input field
   */
  label: string;
  /**
   * Accessibility label for the input
   */
  accessibilityLabel?: string;
};

type TextInputProps = (TextInputPropsWithA11yLabel | TextInputPropsWithLabel) &
  TextInputCommonProps;
```

<img width="692" alt="image" src="https://github.com/razorpay/blade/assets/35374649/07725f7b-bbc3-4b2a-bc6e-9125325b1ecb">

<img width="695" alt="image" src="https://github.com/razorpay/blade/assets/35374649/05b8f438-1d5e-4b9d-8c4d-5206927c5d07">

<img width="622" alt="image" src="https://github.com/razorpay/blade/assets/35374649/d2a7de8b-5f6c-44c8-aa68-632aed665da1">

> if you are planning to support this on all the components you can move these types to the BaseInput instead

Reply:
@anuraghazra but if someone is using Blade with JS they will bypass this check. or by adding `//@ts-ignore` with TS.

Reply:
Updated.

Reply:
> but if someone is using Blade with JS they will bypass this check. or by adding //@ts-ignore with TS.

That's fine. let them bypass. 

```
<labe>
  My custom label
  <Input accessiblityLabel="My custom label" /> // this is redundant can i please ignore this? the label is already set
</label>
```